plugins/woocommerce/src/Admin/API/Reports/Segmenter.php

```diff
@@ -542,6 +542,87 @@ protected function fill_in_missing_interval_segments( &$data ) {
 		}
 	}
 
+	/**
+	 * Calculate segments for totals where the segmenting property is bound to product (e.g. category, product_id, variation_id).
+	 *
+	 * @param array  $segmenting_selections SELECT part of segmenting SQL query--one for 'product_level' and one for 'order_level'.
+	 * @param string $segmenting_from FROM part of segmenting SQL query.
+	 * @param string $segmenting_where WHERE part of segmenting SQL query.
+	 * @param string $segmenting_groupby GROUP BY part of segmenting SQL query.
+	 * @param string $segmenting_dimension_name Name of the segmenting dimension.
+	 * @param string $table_name Name of SQL table which is the stats table for orders.
+	 * @param array  $totals_query Array of SQL clauses for totals query.
+	 * @param string $unique_orders_table Name of temporary SQL table that holds unique orders.
+	 *
+	 * @return array
+	 */
+	protected function get_product_related_totals_segments( $segmenting_selections, $segmenting_from, $segmenting_where, $segmenting_groupby, $segmenting_dimension_name, $table_name, $totals_query, $unique_orders_table ) {
```

Comment:
This method is called by other methods inside this class but it's only defined in subclasses.